pennylane/pauli/conversion.py

```diff
@@ -29,14 +29,10 @@
 
 
 # pylint: disable=too-many-branches
-def pauli_decompose(
-    matrix,
-    hide_identity=False,
-    wire_order=None,
-    pauli=False,
-    padding=False,
-) -> Union[Hamiltonian, PauliSentence]:
-    r"""Decomposes a matrix into a linear combination of Pauli operators.
+def _generalized_pauli_decompose(
+    matrix, hide_identity=False, wire_order=None, pauli=False, padding=False
+) -> Tuple[qml.math.array, list]:
```

Comment:
`qml.math.array` is a partial function, this would probably be better
```suggestion
) -> Tuple[qml.typing.TensorLike, list]:
```